src/components/ResultContainer.tsx

```diff
@@ -58,103 +67,218 @@ function ResultContainer({ response }: { response: QueryResponse | null }) {
     }
   }, [response]);
 
+  function convertURIToLabel(
+    type: string,
+    uri: string | string[] | null
+  ): string | string[] | null {
+    // Handle array of URIs
+    if (Array.isArray(uri)) {
+      return uri.map((singleUri) => convertURIToLabel(type, singleUri)).join(', ');
+    }
+
+    if (!uri) {
+      return uri;
+    }
+
+    switch (type) {
+      case 'sex': {
+        const entry = Object.entries(sexes).find(([, value]) => {
+          const [, id] = value.split(':');
+          return uri.includes(id);
+        });
+        return entry ? entry[0] : uri;
```

Comment:
```suggestion
        return label ? label : uri;
```
see above, I find the `entry[0]` pretty hard to follow

Reply:
OK I see with the Null response type, how about a comment above then to say entry[0] is the label? Not sure there is a way to name this variable better to make it clear without comment. Maybe `matched_pipeline`? But still the same question of what is [0]